cpp/tests/sampling/sampling_heterogeneous_post_processing_test.cpp

```diff
@@ -497,6 +497,8 @@ class Tests_SamplingHeterogeneousPostProcessing
 
         // Check the invariants in vertex renumber_map
 
+        handle.sync_stream();
```

Comment:
We should add a FIXME statement saying something like this shouldn't be necessary but this fixes a segment fault that happens only in specific configurations and this needs further investigation if we want to quickly merge this PR.

Reply:
This needs more investigation (we should better find the root cause of any bugs we encounter when we can reproduce those)... If this manifests later in a much more difficult/expensive to reproduce situations, it can be a nightmare...